src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+In the example below we'll use <CodeStep step={1}>[`navigator.storage.estimate()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate)</CodeStep>, a browser API that provides information on how much storage is being used by the current page and how much space is available. <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is an async function that returns a Promise which resolves once the information is available.
+
+```js [[1,6, "navigator.storage.estimate()"], [2,6, "storagePromise"], [3,6, "useState"], [4,7, "use"], [5,8, "storage.quota"]]
+"use client";
+
+import { use, useState } from "react"
+
+function StorageInfo() {
+  const [storagePromise] = useState(navigator.storage.estimate());
+  const storage = use(storagePromise);
+  return <p>An estimated {storage.quota} bytes available</p>
+}
+```
+
+1. First, the promise returned by <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is stored in the <CodeStep step={2}>`storagePromise`</CodeStep> state variable created by <CodeStep step={3}>`useState`</CodeStep>. This prevents the call to <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> and the promise it returns from being recreated across renders.
+2. Next, <CodeStep step={2}>`storagePromise`</CodeStep> is passed to <CodeStep step={4}>`use`</CodeStep> to read the resolved value of the promise.
+3. Finally, the resolved value of the promise can now be used safely. This includes surfacing data read by <CodeStep step={4}>`use`</CodeStep> like <CodeStep step={5}>`storage.quota`</CodeStep> in the JSX returned by the component.
+
+<Pitfall>
+
+When passing promises to `use`, the promise must be cached with the `cache` function, a state variable, or context to prevent the promise from being recreated on every render.
+
+```js
+function Component() {
+  use(new Promise(...)) // ❌ promise will be recreated on every render
+  // ...
+```
+instead, pass the promise to a state variable or context or wrap the function that returns the promise in `cache`:
+
+```js
+import { use, useState, cache } from "react"
+
+function Component() {
+  const [promise] = useState(() => new Promise(...)) // ✅ Promise persists across renders
+  const data = use(cache(fnThatReturnsAPromise())) // ✅ Promise persists across renders
+  //...
+```
+
+</Pitfall>
+
+
+<Recipes>
+
+#### Read promises from async browser APIs {/*read-promises-from-async-browser-apis*/}
+
+You can read promises returned by async browser APIs like [`navigator.storage.estimate()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate) with `use`. In this example, the promise from `navigator.storage.estimate()` is stored in a state variable and passed to the `StorageInfo` component as a prop. The promise is then read by `use` and the result is displayed to the user. While waiting for the promise to resolve, the fallback of the parent Suspense component is displayed.
```

Comment:
This is duplicated from above. Intentional?

Reply:
It's mentioned `use` can be called conditionally. That could be a great example to include 😄

Reply:
I was thinking that folks might jump to the example, but you're right this is a little wordy. I've thinned it out a bit.

I've added a conditional use example too!